src/benchmarking/genome_similarity/similarity_comparison.py

```diff
@@ -0,0 +1,290 @@
+        kmerLength = genomeLength
+    else:
+        numKmers = genomeLength - kmerLength + 1
+
+    for i in range(numKmers):
+        kmer = genome[i:i + kmerLength]
+        kmers.append(kmer)
+
+    return kmers
+
+def __fasta_to_genome(fasta_path) -> str:
+    ''' 
+        DESCRIPTION:
+            Extracts just the genome from the genome member variable, which should be a file path.
+            Currently assumes only single FASTA file is passed (no multi-fasta). 
+        
+        INPUT:
+            FASTA/FASTQ file (.fa, .fna, .fasta, .fastq)
+        
+        OUTPUT:
+            Genome in string format
+    '''
+    sequence = ""
+    with open(fasta_path) as fasta_file:
+        fasta_lines = fasta_file.readlines()
+        sequence_i = ""
+        for counter, line in enumerate(fasta_lines):
+            if line[0] == ">": 
+                if (counter != 0):
+                    sequence += (sequence_i.rstrip())
+                    sequence_i = ""
+            elif line[0] != ">":
+                sequence += line.rstrip()
+    return sequence
+
+def extract_genome_path(genomes_directory: Path, filename: str) -> list:
+    print(f"Extracting genome paths from: {genomes_directory}")
+
+    return Path(genomes_directory) / Path(filename)
+
+
+def populate_genome_dict(genomes_directory: Path) -> dict:
+    '''
+    Dictionary: [ fullpath : genome str]
+    '''
+    print(f'Populating genome dictionary from: {genomes_directory}')
+    genomeDict = {}
+    for genome_filename in os.listdir(genomes_directory):
+        if genome_filename.endswith('fa') or genome_filename.endswith('fna') or genome_filename.endswith('fasta'):
+            fullpath = Path(genomes_directory) / Path(genome_filename)
+            genomeDict[fullpath] = __fasta_to_genome(fullpath)
+
+    return genomeDict
+
+
+def run_jaccard(genomePair: list, kmerLength=35) -> float:
+    '''
+    Creates kmers out of the provided genomes to calculate the jaccard index between
+    the sets of kmers
+
+    INPUT:
+        genomePair: list of (2) genomes
+    '''
+    kmerLength=int(kmerLength)
+    genome1_kmers = __create_kmers(__fasta_to_genome(genomePair[0]), kmerLength)
+    genome2_kmers = __create_kmers(__fasta_to_genome(genomePair[1]), kmerLength)
+    
+    a = set(genome1_kmers)
+    b = set(genome2_kmers)
+
+    intersection = len(a.intersection(b))
+    union = len(a.union(b))
+    
+    return round((intersection / union)*100, 4)
+
+
+def run_dnadiff(genomePair: list, outputdir: str):
+    print('Running dnadiff')
+    subprocess.run(['dnadiff', '-p', Path(outputdir)/Path('dnadiff_out'), genomePair[0], genomePair[1]])
+    #return Path(outputdir + '/dnadiff_out.report')
+    similarity=-1.0
+    with open(outputdir+ '/dnadiff_out.report') as dnadiff_file:
+        for line in dnadiff_file:
+            if 'AvgIdentity' in line:
+                similarity = line.rstrip().split()[1]
+                break; # only first occurrence
+    return similarity
```

Comment:
@latifa-alnaimi 

I know this is just a script for getting things done, but a cool design pattern to use here would be an adapter pattern. Imagine a class that has a genome compare method and returns similarity, such as this method. Another method could encapsulate how the subprocess is happening, and perhaps as a singleton since only one is needed per time.